### ISSUE TITLE
Auto number creation problem

### ISSUE THREAD

**Issue by mortoza** [tyl/field-binder#5]
When I am using     @AutoGenerated, getting exceptions.

Person2 bean

```
@Id
@AutoGenerated
private Long id;
private String firstName;
private String middleName;
private String lastName;
```

exception:

Caused by: javax.persistence.PersistenceException: Exception [EclipseLink-4002](Eclipse Persistence Services - 2.5.2.v20140319-9ad6abd): org.eclipse.persistence.exceptions.DatabaseException
Internal Exception: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column 'Id', table 'inteacc.dbo.Person2'; column does not allow nulls. INSERT fails.
Error Code: 515
Call: INSERT INTO PERSON2 (ID, AGE, BIRTHDATE, FIRSTNAME, LASTNAME, MIDDLENAME) VALUES (?, ?, ?, ?, ?, ?)
    bind => [6 parameters bound]
Query: InsertObjectQuery(Person{firstName='ghgh', middleName='ghgjk', lastName='kll'})
    at org.eclipse.persistence.internal.jpa.EntityManagerSetupImpl$1.handleException(EntityManagerSetupImpl.java:696)
    at org.eclipse.persistence.transaction.AbstractSynchronizationListener.handleException(AbstractSynchronizationListener.java:275)
    at org.eclipse.persistence.transaction.AbstractSynchronizationListener.beforeCompletion(AbstractSynchronizationListener.java:170)
    at org.eclipse.persistence.transaction.JTASynchronizationListener.beforeCompletion(JTASynchronizationListener.java:68)
    at com.sun.enterprise.transaction.JavaEETransactionImpl.commit(JavaEETransactionImpl.java:452)
    ... 81 more
Caused by: Exception [EclipseLink-4002](Eclipse Persistence Services - 2.5.2.v20140319-9ad6abd): org.eclipse.persistence.exceptions.DatabaseException
Internal Exception: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert the value NULL into column 'Id', table 'inteacc.dbo.Person2'; column does not allow nulls. INSERT fails.
Error Code: 515
Call: INSERT INTO PERSON2 (ID, AGE, BIRTHDATE, FIRSTNAME, LASTNAME, MIDDLENAME) VALUES (?, ?, ?, ?, ?, ?)
    bind => [6 parameters bound]
Query: InsertObjectQuery(Person{firstName='ghgh', middleName='ghgjk', lastName='kll'})
    at org.eclipse.persistence.exceptions.DatabaseException.sqlException(DatabaseException.java:331)

**Comment by evacchi** [comment]
The JPA Crud implementation is very simple. The `onCommit()` event is implemented as follows:

``` java
        DataNavigation nav = event.getSource();
        T bean = fieldBinder.getBeanDataSource();
        JPAContainer<T> container = (JPAContainer<T>) nav.getContainer();
        Object id = container.addEntity(bean);
        nav.setCurrentItemId(id);
```

the `onCreate()` event is even simpler:

``` java
            event.getSource().setCurrentItemId(null);
            T bean = createBean(); // equivalent to new Person2()
            fieldBinder.setBeanDataSource(bean);
```

which is, as far as I know, the usual JPA workflow. You may need to customize these events if your model differs. The `addEntity()` method [is the suggested way to add an entity in a JPAContainer.](https://vaadin.com/book/-/page/jpacontainer.usage.html). The method usually does a `merge()` of the fresh entity instance.

If this may be useful, in my tests the `Person` entity's `id` is defined as follows:

``` java
    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    private long id;
```

which works as intended:

![Imgur](http://i.imgur.com/zBXTI2F.png)

**Comment by mortoza** [comment]
Hi
While I am using the code block in OnCommit, see import of T is not recognized, is there something I am missing here, Do I need to declare T? Sorry for this novice question!

  binder.getNavigation().addOnCommitListener(new OnCommit.Listener() {
            @Override
            public void onCommit(OnCommit.Event event) {
                DataNavigation nav = event.getSource();
                T bean = binder.getBeanDataSource();
                JPAContainer<T> container = (JPAContainer<T>) nav.getContainer();
                Object id = container.addEntity(bean);
                nav.setCurrentItemId(id);  
            }
        });

**Comment by evacchi** [comment]
I guess I did not explain myself well. The code I pasted is code from the _built-in_ implementation in the FieldBinder. You do not have to write it yourself (the T type was a generic type in the built-in implementation). My point was to show you that the code of the JPA implementation is rather simple. 

The only thing you might need to play with is your JPA configuration. For instance,     

``` java
    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    private long id;
```

is working for me
